### DISCUSSION TITLE
A way to set the default (chat) agent?

### DISCUSSION THREAD

**Discussion by dnaaun** [Robitx/gp.nvim#85] [Q&A] [answered]
I couldn't find a way to do this, so I'm using the following:

```lua
		-- I don't know of an official way to set the default agent on startup, so I use
		-- an undocumented/internal API:
		require("gp")._state.chat_agent = "ChatGPT4"
```

**Comment by Robitx** [comment, marked as answer]
@dnaaun Hey, is there a need for default when the [last setting](https://github.com/Robitx/gp.nvim?tab=readme-ov-file#agent-commands) is persisted on disk? :slightly_smiling_face:

**Reply by dnaaun** [reply]
Sorry for bailing here, but yeah, if that works, that's great! I'm going to stick with my solution until my config breaks. Given that this is (neo)vim, I don't think that will take long 😛

**Comment by edshamis** [comment]
It would be nice to have this in config for syncing between devices

**Comment by alex-popov-tech** [comment]
agree with @edshamis , would be nice to have it in config

for some reason, mine do not save this setting on disc :(

**Reply by alex-popov-tech** [reply]
@Robitx can you please consider having this in config? or will you accept PR with such change?

**Comment by Robitx** [comment]
@alex-popov-tech @edshamis @dnaaun in the latest version:
```lua
	-- default agent names set during startup, if nil last used agent is used
	default_command_agent = nil,
	default_chat_agent = nil,
```
